components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -229,47 +229,43 @@ func (g *rtkSerial) start() error {
 
 // getStream attempts to connect to ntrip stream. We give up after maxAttempts unsuccessful tries.
 func (g *rtkSerial) getStream(mountPoint string, maxAttempts int) error {
-	g.mu.Lock()
-	defer g.mu.Unlock()
+	success := false
+	attempts := 0
 
 	var rc io.ReadCloser
 	var err error
 
 	g.logger.Debug("Getting NTRIP stream")
 
-	for attempts := 0; attempts < maxAttempts; attempts++ {
-		if g.isClosed {
-			return g.err.Get()
-		}
-
+	for !success && attempts < maxAttempts {
```

Comment:
Susmita and I discussed: the trouble with the old version was that the mutex was locked for the _entirety_ of this for loop, which is many times longer than the slow line (`g.ntripClient.Client.GetStream()`). _That_ was the slow part: doing the slow thing many times in a loop, rather than just doing it once.